### ISSUE TITLE
Copyright Date

### ISSUE THREAD

**Issue by williamkapke** [nodejs/TSC#195]
Hopefully someone already knows (maybe @mikeal?) the answer to this and it's easy...

When we have one of these:
> Copyright (c) 2016 NAN contributors

Are we supposed to go through and change the dates on all of these?

**Comment by ljharb** [comment]
I don't think it's supposed to be updated, generally, even though most do it. http://stackoverflow.com/questions/2390230/do-copyright-dates-need-to-be-updated

**Comment by Fishrock123** [comment]
> I don't think it's supposed to be updated, generally, even though most do it. 

Isn't the "most correct" way to do it like "(c) 2012-2017" then? Or am I mistaken?

**Comment by ljharb** [comment]
If you're going to update it, yes. But I also don't think it needs to be updated until the copyright is due to expire maybe?

**Comment by williamkapke** [comment]
Some places uses in our org do not have a date at all. What does THAT mean then? 🤔 

Anyhow, I've added the `tsc-agenda` label to line us up to discuss and then MAYBE asking the board to ask the legal committee for an official answer.

**Comment by Trott** [comment]
I Am Not A Lawyer. That said:

* Lots of reputable sites don't put a year at all, which makes sense to me in the context of the web.

* For that matter, lots of reputable sites don't include a copyright notice whatsoever (e.g., Wikipedia, Yahoo!, Google...).

* You should not update the copyright year on a document that has not changed. But, of course, everyone does that.

(FWIW, I'd recommend kicking it up to the board and not even bothering discussing at the TSC level.)

**Comment by mikeal** [comment]
This is the best explainer I know of https://writing.kemitchell.com/2016/09/21/MIT-License-Line-by-Line.html

TLDR; the date and notice are not actually necessary but *do* have to be preserved in whatever state the creator sets them under the license terms :)

**Comment by rvagg** [comment]
@mikeal @ TSC meeting today:

> Was discussed at the board meeting, more complicated than I originally thought but will need to consult the notes to see what the discussion was

**Comment by jasnell** [comment]
There are several fundamental considerations here:

1. For *existing* copyright notices on Node.js Foundation projects:
  * Is it ok to *update* the year included in the copyright (if any) without Legal Committee approval?
  * Is it ok to *remove* the year included in the copyright (if any)

2. For new projects, we should likely have a standard Node.js Foundation copyright notice that is standardized and approved by the Foundation Board. This should become part of the project governance policy.

Item 2 is a larger issue that will need to be handled separately. I will draft up a separate proposal for that. What we need the board to decide on more immediately is Item 1.

**Comment by mhdawson** [comment]
@jasnell any progress on putting together the proposal

**Comment by jasnell** [comment]
I've got an open issue. From what I understand there's been zero movement on the foundation legal committee side of things. I'll follow up again next week

**Comment by maclover7** [comment]
ping @MylesBorins @jasnell since we're getting close to new year's

**Comment by Trott** [comment]
Just want to make sure this doesn't get lost: It would be awesome if we don't need to keep a date in there at all. Just a copyright notice with no year. Lots of reputable sites do this. But I-Am-Not-A-Lawyer.

**Comment by mhdawson** [comment]
@MylesBorins I think we were waiting on a response from the Foundation legal committee, can you chase that   ?

**Comment by jasnell** [comment]
Recommending that we just close this given that there's been zero activity and movement.

**Comment by MylesBorins** [comment]
This is still being discussed by the legal team afaik. Maybe @mrhinkle can
give us an update on the 28th meeting

**Comment by bnb** [comment]
@MylesBorins I commented in https://github.com/nodejs/community-committee/issues/245, but the CommComm has several licenses pending on legal feedback and @williamkapke referenced this issue, though I think copyright discussed in this issue is likely different than licensing 🤔

**Comment by mrhinkle** [comment]
The Board Foundation Subcommittee met last month and the board discussed this issue in our last meeting. Here's the summary of the discussion. The net of the conversation is that the copyright notice is not critical but the best practices were boiled down to the following:

- Do not strip or alter existing copyright notices. 
- Other projects have utilized “[Foundation Name or Project Name] and its contributors”.
- Adding dates is not preferable.  It does not have a legal impact and would be excluded only for ease of administration.
- If a file is updated by the project past the original author, one may add the project copyright notice.
- SPDX tagging (https://spdx.org/) can be helpful for scanning when adding new files

**Comment by fhinkel** [comment]
@mrhinkle thanks for clarifying. 

Closing this, feel free to reopen if needed.
